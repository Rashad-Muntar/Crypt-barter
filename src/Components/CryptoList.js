/* eslint-disable no-use-before-define, consistent-return, arrow-body-style */
import React, { useEffect } from 'react';
import '../Assets/cryptoList.css';
import { useDispatch, useSelector } from 'react-redux';
import { Link } from 'react-router-dom';
import { filterAction, coinsFetcher } from '../Actions/index';
// import CryptoDetails from './CryptoDetails';
import FilterForm from './FilterForm';
import Crypto from './Cryto';

const CryptoList = () => {
  const coinsData = useSelector((state) => state.cryptoReducer.data);
  const filter = useSelector((state) => state.filterReducer);
  console.log(filter);
  console.log(coinsData);
  const dispatch = useDispatch();

  const handleFilterChanger = (e) => {
    dispatch(filterAction(e.target.value));
  };

  useEffect(() => {
    dispatch(coinsFetcher());
  }, []);

  const filteredCryptos = (filter !== '') ? coinsData.coins.filter((coin) => coin.name === filter) : coinsData.coins;
  return (
    <div className="categoryLisWrapper">
      <FilterForm onCrytoFilter={handleFilterChanger} />
      {
      filteredCryptos.map((coin) => {
        return (
          <Link to={`/details/${coin.id}`} key={coin.id}>
            <Crypto
              key={coin.id}
              name={coin.name}
              price={coin.price}
            />
          </Link>
        );
      })
    }

    </div>
  // coinsData.data.map((coin) => (
  //   <Crypto
  //     key={coin.id}
  //     name={coin.name}
  //     price={coin.price}
  //   />
  // ))
  );
};

// CryptoList.propTypes = {
//     book: PropTypes.string.isRequired,
//     title: PropTypes.string.isRequired,
//     handleRemoveBook: PropTypes.func.isRequired,
//   };

export default CryptoList;
