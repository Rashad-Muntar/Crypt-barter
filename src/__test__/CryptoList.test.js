import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import CryptoList from '../Components/Home';
import rootReducer from '../Reducers/index';

it('renders correctly when store is provided', () => {
  const store = createStore(rootReducer);
  const cryptoList = renderer.create(
    <Provider store={store}>
      <CryptoList />
    </Provider>,
  ).toJSON();
  expect(cryptoList).toMatchSnapshot();
});
