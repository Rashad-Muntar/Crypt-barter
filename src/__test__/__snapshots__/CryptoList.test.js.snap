// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when store is provided 1`] = `
<div
  className="categoryLisWrapper"
>
  <div
    className="image-section"
  >
    <div
      className="search"
    >
      <i
        className="search icon"
      />
      <input
        onChange={[Function]}
        placeholder="Enter crypto name"
        type="text"
      />
    </div>
  </div>
  <div
    className="cards-wrapper"
  />
</div>
`;
